File: app/src/main/java/doit/study/droid/data/Question.java
package doit.study.droid.data;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;

import java.util.List;

import static doit.study.droid.data.Converters.listToString;

@Entity(tableName = "questions")
public class Question {
    @PrimaryKey
    public final int id;
    public final String text;
    public final String wrong;
    public final String right;
    public final String docLink;

    public Question(int id, String text, String wrong, String right, String docLink) {
        this.id = id;
        this.text = text;
        this.wrong = wrong;
        this.right = right;
        this.docLink = docLink;
    }

    public Question(int id, String text, List<String> wrong, List<String> right, String docLink) {
        this.id = id;
        this.text = text;
        this.wrong = listToString(wrong);
        this.right = listToString(right);
        this.docLink = docLink;
    }
}

File: app/src/main/java/doit/study/droid/data/local/entity/Question.kt
package doit.study.droid.data.local.entity

import androidx.annotation.NonNull
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.*

@Entity
data class Question (
        @PrimaryKey
        @NonNull
        // Read only
        val id: Int,
        val text: String,
        val wrong: List<String>,
        val right: List<String>,
        val docLink: String,
        // Read/Write
        val wrongCounter: Int = 0,
        val rightCounter: Int = 0,
        val consecutiveRightCounter: Int = 0,
        val lastViewedAt: Long = Date().time,
        val studiedAt: Long = 0
)
